tailcfg/tailcfg.go

```diff
@@ -1097,6 +1097,9 @@ type RegisterRequest struct {
 	Timestamp     *time.Time    `json:",omitempty"` // creation time of request to prevent replay
 	DeviceCert    []byte        `json:",omitempty"` // X.509 certificate for client device
 	Signature     []byte        `json:",omitempty"` // as described by SignatureType
+
+	// The following fields are used for generating login urls given mdm and non-mdm required clients.
```

Comment:
URLs, MDM, non-MDM

But what _is_ it? Or is this just the heading for the following fields. If so, add a blank line after it so it's not touching the Tailnet field.

But then document the Tailnet field in the normal https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences style.

```
// Tailnet, if non-empty, is the Tailnet name (the same form used
// in the Tailscale API for ....
```

But https://github.com/tailscale/tailscale/blob/main/api.md#tailnet and https://login.tailscale.com/admin/settings/general call it the "Organization name", weirdly.

@maisem, you know where this naming is going?

Reply:
`MapResponse.Domain` uses this language:
```golang
	// Domain is the name of the network that this node is
	// in. It's either of the form "example.com" (for user
	// foo@example.com, for multi-user networks) or
	// "foo@gmail.com" (for siloed users on shared email
	// providers). Its exact form should not be depended on; new
	// forms are coming later.
	// If empty, the value is unchanged.
	Domain string `json:",omitempty"`
```

As much as it pains me to add `Domain` to refer to this in another place, maybe staying consistent with that would make sense? Another possible term would be `Tenant` but I don't think we use that anywhere currently and adding another name to the mix seems like a bad idea.

Maybe:

```golang
	// Tailnet is an optional identifier specifying the name of the recommended or required
	// network that the node should join. Its exact form should not be depended on; new
	// forms are coming later. The identifier is generally a domain name (for an organization)
	// or e-mail address (for a personal account on a shared e-mail provider). It is the same name
	// used by the API, as described in /api.md#tailnet.
	// If Tailnet begins with the prefix "required:" then the server should prevent logging in to a different
	// network than the one specified. Otherwise, the server should recommend the specified network
	// but still permit logging in to other networks.
	// If empty, no recommendation is offered to the server and the login page should show all options.
	Tailnet string `json:",omitempty"`
```